rpcd/playbooks/roles/rpc_maas/tasks/package_install.yml

```diff
@@ -21,9 +21,27 @@
     cache_valid_time: 600
   with_items: maas_apt_packages
 
-- name: Install pip packages
+- name: Create venv dir for maas
+  file: path={{ maas_venv }} state=directory
+
+- name: Copy pip conf to venv
+  copy: src=/root/.pip/pip.conf dest="{{ maas_venv }}/pip.conf"
+
+- name: Update pip
+  pip:
+    name: pip
+    extra_args: '--upgrade'
+    virtualenv: "{{ maas_venv_enabled | bool | ternary(maas_venv_bin|dirname,omit) }}"
+
+- name: Install pip packages to venv
   pip:
     name: "{{ item }}"
-    state: latest
-    extra_args: "{{ pip_install_options | default('') }}"
+    state: present
+    extra_args: "{{ pip_install_options | default('-vvv') }}"
+    virtualenv: "{{ maas_venv_enabled | bool | ternary(maas_venv_bin|dirname,omit) }}"
+    virtualenv_site_packages: "{{ maas_venv_enabled | bool | ternary('no', omit) }}"
+  register: install_pip_packages
+  until: install_pip_packages|success
+  retries: 5
```

Comment:
No, the delay is on the next line,  my point is that we should not need to retry.  we tell it to install xyz pip package and it should do so the first time, no need to retry.